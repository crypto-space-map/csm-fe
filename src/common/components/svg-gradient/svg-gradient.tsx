import styled from '@emotion/styled';
import { COLOR_PALLETTE } from 'global/pallette';

const { MAIN_BLUE, MAIN_GREEN } = COLOR_PALLETTE;

const Svg = styled.svg`
  position: fixed;
  z-index: -1;
  width: 0;
  height: 0;
  fill: none;
`;

/**
 * Use csm_icon_gradient id to add ico fill like (  fill: url(#csm_icon_gradient)   )
 */
export const SvgGradient = () => (
  <Svg viewBox="0 0 0 0">
    <defs>
      <linearGradient
        id="csm_icon_gradient"
        x1="18"
        y1="8.01351e-07"
        x2="-2.75068"
        y2="13.891"
        gradientUnits="userSpaceOnUse">
        <stop stopColor={MAIN_GREEN} />
        <stop offset="1" stopColor={MAIN_BLUE} />
      </linearGradient>
    </defs>
  </Svg>
);
