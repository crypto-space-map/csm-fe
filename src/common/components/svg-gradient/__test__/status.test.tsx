import { render } from '@testing-library/react';

import { SvgGradient } from '../svg-gradient';

describe('<SvgGradient />', () => {
  it('should match snapshot', () => {
    const { container } = render(<SvgGradient />);
    expect(container).toMatchSnapshot();
  });
  it('should be defined', () => {
    const { container } = render(<SvgGradient />);
    expect(container).toBeDefined();
  });
});
