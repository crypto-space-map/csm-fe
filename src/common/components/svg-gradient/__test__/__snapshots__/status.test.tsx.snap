// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SvgGradient /> should match snapshot 1`] = `
<div>
  <svg
    class="css-57cqf"
    viewBox="0 0 0 0"
  >
    <defs>
      <lineargradient
        gradientUnits="userSpaceOnUse"
        id="csm_icon_gradient"
        x1="18"
        x2="-2.75068"
        y1="8.01351e-07"
        y2="13.891"
      >
        <stop
          stop-color="#a8f8a0"
        />
        <stop
          offset="1"
          stop-color="#83d9f5"
        />
      </lineargradient>
    </defs>
  </svg>
</div>
`;
